src/lib.rs

```diff
@@ -597,13 +584,58 @@ pub mod pallet {
             })
         }
 
+        /// Convert Move address to Substrate native account.
+        pub fn to_native_account(address: &AccountAddress) -> Result<T::AccountId, Error<T>> {
+            T::AccountId::decode(&mut address.as_ref()).map_err(|_| Error::InvalidAccountSize)
+        }
+
+        /// Convert a native address to a Move address.
+        pub fn to_move_address(address: &T::AccountId) -> Result<AccountAddress, Error<T>> {
+            let address = AccountId32::decode(&mut address.encode().as_ref())
+                .map_err(|_| Error::InvalidAccountSize)?;
+
+            let account_bytes: [u8; 32] = address.into();
+            Ok(AccountAddress::new(account_bytes))
+        }
+
+        /// Publish the module using the appropriate gas strategy.
+        pub fn raw_publish_module(
+            address: &T::AccountId,
+            bytecode: Vec<u8>,
+            gas: GasStrategy,
+        ) -> Result<VmResult, Error<T>> {
+            let storage = Self::move_vm_storage();
+            let vm = Mvm::new(storage).map_err(|_err| Error::<T>::PublishModuleFailed)?;
+            let address = Self::to_move_address(address).unwrap();
```

Comment:
That can never fail?

Maybe that could be better:
```suggestion
            let address = Self::to_move_address(address)?;
```

Reply:
Done